cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -1197,173 +1196,69 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
                 runInference={async (model: MLModel, body: DetectorRequestBody) => {
                     function loadAttributes(
-                        attributes: { name: string; value: string }[],
-                        label: Label,
+                        attributes: { spec_id: number; value: string }[],
                     ): Record<number, string> {
-                        return attributes.reduce((acc, { name, value }) => {
-                            const attributeSpec = label.attributes.find((_attr) => _attr.name === name);
-
-                            if (!attributeSpec) {
-                                return acc;
-                            }
-
-                            switch (attributeSpec.inputType) {
-                                case 'number': {
-                                    const [min, max, step] = attributeSpec.values;
-                                    if (
-                                        Number.isFinite(+value) &&
-                                        +value >= +min &&
-                                        +value <= +max &&
-                                        !(+value % +step)
-                                    ) {
-                                        return {
-                                            ...acc,
-                                            [attributeSpec.id as number]: `${value}`,
-                                        };
-                                    }
-
-                                    return acc;
-                                }
-                                case 'text': {
-                                    return {
-                                        ...acc,
-                                        [attributeSpec.id as number]: `${value}`,
-                                    };
-                                }
-                                case 'select':
-                                case 'radio': {
-                                    if (attributeSpec.values.includes(value)) {
-                                        return {
-                                            ...acc,
-                                            [attributeSpec.id as number]: value,
-                                        };
-                                    }
-                                    return acc;
-                                }
-                                case 'checkbox':
-                                    return {
-                                        ...acc,
-                                        [attributeSpec.id as number]: `${value}`.toLowerCase() === 'true' ? 'true' : 'false',
-                                    };
-                                default:
-                                    return acc;
-                            }
-                        }, {} as Record<number, string>);
+                        return Object.fromEntries(attributes.map((a) => [a.spec_id, a.value]));
```

Comment:
We lost min/max/step validation for numeric attributes for interactive requests
Server only checks .isnumeric() during validation

BTW, one more potential issue in related code is: "0.5".isnumeric() returns **False**, thus such values returned from a serverless functions are ignored (e.g. confidence)

Reply:
> We lost min/max/step validation for numeric attributes for interactive requests
> Server only checks .isnumeric() during validation

Ah, good point. I added range checking to the server validation logic. I also found out that the step checking was incorrect (it checked "value divisible by step" instead of "(value - min) divisible by step"), so I fixed that as well.

> BTW, one more potential issue in related code is: "0.5".isnumeric() returns False, thus such values returned from a serverless functions are ignored (e.g. confidence)

I'm pretty sure that confidence is ignored regardless (I don't even know why Nuclio functions return it...). Aside from that, I don't think attributes can have fractional values anyway, can they? Such values would fail the step check.